sippy-ng/src/component_readiness/IncludeVariantCheckboxList.js

```diff
@@ -1,31 +1,190 @@
+    setCompareItems(newItems)
+  }
+
+  const handleVariantSelection = (
+    event,
+    checkedItems,
+    setCheckedItems,
+    updateCheckedItems
+  ) => {
+    const item = event.target.name
+    const newItems = event.target.checked
+      ? [...checkedItems, item]
+      : checkedItems.filter((checkedItem) => checkedItem !== item)
+    setCheckedItems(newItems)
+    updateCheckedItems(newItems)
+  }
+  const handleIncludeVariantSelection = (event) => {
+    handleVariantSelection(
+      event,
+      includeItems,
+      setIncludeItems,
+      updateIncludeItems
+    )
+  }
+  const handleCompareVariantSelection = (event) => {
+    handleVariantSelection(
+      event,
+      compareItems,
+      setCompareItems,
+      updateCompareItems
+    )
+  }
+
+  const [isCompareMode, setIsCompareMode] = useState(
+    varsContext.variantCrossCompare.includes(variantGroupName)
+  )
+  const handleToggleCompare = (event) => {
+    varsContext.updateVariantCrossCompare(variantGroupName, !isCompareMode)
+    setIsCompareMode(!isCompareMode)
+  }
+
+  const classes = makeStyles((theme) => ({
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: '20px',
+    },
+    gridCenter: {
+      marginTop: theme.spacing(1),
+      textAlign: 'center',
+    },
+    gridRight: {
+      justifyContent: 'flex-end',
+      display: 'flex',
+    },
+  }))()
 
-  const updateCheckedItems = (newCheckedItems) => {
-    varsContext.replaceIncludeVariantsCheckedItems(variantName, newCheckedItems)
-    setCheckedItems(newCheckedItems)
+  let params = {
+    variantGroupName,
+    classes,
+    displayList: varsContext.allJobVariants[variantGroupName],
+    includeItems,
+    handleIncludeVariantSelection,
+    compareItems,
+    handleCompareVariantSelection,
   }
+
+  return (
+    <FormControl
+      variant="standard"
+      className={classes.formControl}
+      component="fieldset"
+    >
+      <Accordion className="checkboxlist-headerName">
+        <AccordionSummary
+          className="checkboxlist-summary"
+          expandIcon={<ExpandMore />}
+        >
+          <Typography className="checkboxlist-label">
+            Include {variantGroupName}
+          </Typography>
+        </AccordionSummary>
+        <AccordionDetails>
+          {isCompareMode
+            ? CompareVariantGroup(params)
+            : IncludeVariantGroup(params)}
+        </AccordionDetails>
+      </Accordion>
+    </FormControl>
+  )
+}
+
+function IncludeVariantGroup(params) {
+  return (
+    <FormGroup>
+      {params.displayList.map((item) => (
+        <FormControlLabel
+          key={item}
+          control={
+            <Checkbox
+              checked={params.includeItems.includes(item)}
+              onChange={params.handleIncludeVariantSelection}
+              name={item}
+            />
+          }
+          label={item}
+        />
+      ))}
+    </FormGroup>
+  )
+}
+
+function CompareVariantGroup(params) {
```

Comment:
This may depend on what you come up with for server side, but is there a plan yet for single vs multi-select? I suspect if you enable this compare mode you must be restricted to one value on each side for us to know what to compare to what.

Reply:
I am hoping that there will be no need to restrict it to one per side. You'll simply have basis variants and sample variants (e.g. IBM vs non-IBM arches). Tracing through the logic with Forrest yesterday, the only place where this seemed to break down is when referencing triage data, which could specify any variants, while the basis/sample would come back with data not grouped by the cross-compared variants, so we can't always correlate the triage data (while we could if sample/basis only had one variant each). But we concluded as a first approximation that for cross-compare scenarios, triage data may not be important.

Reply:
the side-by-side job comparison on test details will also be messy, since the job names could vary fairly wildly across variants. but that would really be the case whether there were single or multiple variants on each side, and again as a first pass i think we'll still have something useful.

Reply:
On test details I think that's fine if virtually no jobs are matched, it's really about the overall data and links to job runs.

If you can pull this off with multi comparing to multi that sounds awesome.